test/test_nn.py

```diff
@@ -16404,6 +16404,81 @@ def _helper(zero_infinity):
         _helper(zero_infinity=True)
         _helper(zero_infinity=False)
 
+    def _CTCLoss_no_batch_dim_helper(self, device):
+        input_length = 40
+        vocab_size = 3
+        batch_size = 1
+        target_length = 12
+
+        log_probs = torch.randn(input_length, batch_size, vocab_size, dtype=torch.float, device=device) \
+                         .log_softmax(2).requires_grad_()
+        targets = torch.randint(low=1, high=vocab_size - 1, size=(batch_size, target_length), dtype=torch.int, device=device)
+        input_lengths = batch_size * [input_length]
+        target_lengths = batch_size * [target_length]
+
+        log_probs_no_bd = log_probs.squeeze(1)
+        targets_no_bd = targets.squeeze(0)
+        input_lengths_no_bd = torch.tensor(input_length)
+        target_lengths_no_bd = torch.tensor(target_length)
+
+        return ((log_probs, targets, input_lengths, target_lengths),
+                (log_probs_no_bd, targets_no_bd, input_lengths_no_bd, target_lengths_no_bd))
+
+    @onlyCUDA
+    @skipCUDAIfNoCudnn
+    def test_CTCLoss_no_batch_dim_cudnn(self, device):
+        args, args_no_bd = self._CTCLoss_no_batch_dim_helper(device)
+        log_probs, targets, input_lengths, target_lengths = args
+        log_probs_no_bd, targets_no_bd, input_lengths_no_bd, target_lengths_no_bd = args_no_bd
+
+        with torch.backends.cudnn.flags(enabled=False):
+            res = torch.nn.functional.ctc_loss(log_probs, targets, input_lengths, target_lengths, zero_infinity=True)
+            res_no_bd = torch.nn.functional.ctc_loss(log_probs_no_bd, targets_no_bd, input_lengths_no_bd,
+                                                     target_lengths_no_bd, zero_infinity=True)
+
+        grad_out = torch.randn_like(res)
+        grad, = torch.autograd.grad(res, log_probs, grad_out)
+        grad_out_no_bd = grad_out.detach().clone().double().requires_grad_()
+        grad_no_bd, = torch.autograd.grad(res_no_bd, log_probs_no_bd, grad_out_no_bd)
+
+        with torch.backends.cudnn.flags(enabled=True):
+            res_mixed = torch.nn.functional.ctc_loss(log_probs, targets.cpu(), input_lengths, target_lengths, zero_infinity=True)
+            res_no_bd_mixed = torch.nn.functional.ctc_loss(
```

Comment:
I think it's slightly better to test the module form of `CTCLoss` in this test and below as this will also test the functional form.